src/EnergyPlus/UnitarySystem.cc

```diff
@@ -18058,5 +18092,147 @@ namespace UnitarySystems {
             });
     }
 
+    Real64 UnitarySys::getFanDeltaTemp(EnergyPlusData &state, bool const firstHVACIteration, Real64 const massFlowRate, Real64 const airFlowRatio)
+    {
+        Real64 fanDT = 0;
+        int FanInletNode = 0;
+        int FanOutletNode = 0;
+        if (this->m_FanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
+            FanInletNode = state.dataHVACFan->fanObjs[this->m_FanIndex]->inletNodeNum;
+            FanOutletNode = state.dataHVACFan->fanObjs[this->m_FanIndex]->outletNodeNum;
+            state.dataLoopNodes->Node(FanInletNode).MassFlowRate = massFlowRate;
+            state.dataHVACFan->fanObjs[this->m_FanIndex]->simulate(state, airFlowRatio, _, _, _, _, _);
+        } else {
+            FanInletNode = state.dataFans->Fan(this->m_FanIndex).InletNodeNum;
+            FanOutletNode = state.dataFans->Fan(this->m_FanIndex).OutletNodeNum;
+            state.dataLoopNodes->Node(FanInletNode).MassFlowRate = massFlowRate;
+            Fans::SimulateFanComponents(state, blankString, firstHVACIteration, this->m_FanIndex, airFlowRatio);
+        }
+        fanDT = state.dataLoopNodes->Node(FanOutletNode).Temp - state.dataLoopNodes->Node(FanInletNode).Temp;
+        return fanDT;
+    }
```

Comment:
New function to calculate a fan's node temperature differential based on a specific mass flow and air flow ratio.